### ISSUE TITLE
Unable to work with vars in bash script

### ISSUE THREAD

**Issue by larkoie** [mikefarah/yq#1122]
Hi,
I'm trying to use your very cool tool to bulk modify yaml files but i can't make it work with both key and value as a var.
I have a txt file with all the keys in it which i read line by line in the script to copy the key from the source file and past it in the destination file.
I use a for loop but also tried with a while and both results are the same : no error but also no changes in the destination file:

```
for i in $lines
do
  valueName=$(yq ${i} ${sourceFile})
  key="$i"
  yq -i 'env(key) = env(valueName)' $destFile
done
```

I tried lots of variation from answers you gave in other issues from this github but none is working unfortunately. I did echo both vars (key and valueName) and both are populated with the correct values.
I'm stuck and can't understand what's wrong with this as the vars are declared in the for loop so it should work right?

the closest working thing i could produce is when the key patch is hardcoded and the value is interpreted in a bash var :
`yq -i '.my.key.path = "'${valueName}'"' $destfile
`
If you have an explanation for this, i'll be glad to read it :)

Thank you

**Comment by mikefarah** [comment]
Hi - for the key, you need to tell `yq` that is a path to something, otherwise where you're doing is telling it to assign a the scalar value of valueName to the scalar value of key.

See https://mikefarah.gitbook.io/yq/operators/env-variable-operators#dynamic-key-lookup-with-environment-variable for more information.

TLDR you need to wrap key in `.[env(key)]` so it knows its a path to something.

```
 yq -i '.[env(key)] = env(valueName)' $destFile
```

**Comment by larkoie** [comment]
thank you for the precision Mike, appreciate it.
